server/src/main/java/server/logger/BazelServerLogger.java

```diff
@@ -0,0 +1,28 @@
+package server.logger;
+
+
+import java.io.PrintStream;
+
+public class BazelServerLogger {
+
+    private static BazelServerLogger instance;
+
+    private static PrintStream logStream = System.out;
+
+    static void setStream(PrintStream stream) {
+        logStream = stream;
+    }
+
+    public static BazelServerLogger getLogger() {
+        if (instance == null) {
+            instance = new BazelServerLogger();
+        }
+        return instance;
+    }
+
+    public void log(String line) {
```

Comment:
I wonder if we ought to make each method here static? So instead of

```
BazelServerLogger.getLogger().log('hello')
```

We could do

```
BazelServerLogger.log('hello')
```

Only reason being that it's shorter and I imagine we'll be using this class a lot